python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -523,14 +523,19 @@ def select_operator_to_run(
     ops = []
     for op, state in topology.items():
         under_resource_limits = _execution_allowed(op, resource_manager)
+        in_backpressure = (
+            any(not p.can_add_input(op) for p in backpressure_policies)
+            or not under_resource_limits
```

Comment:
nit, swap the order of these 2 conditions. so when `not under_resource_limits`, we don't need to go over the policies.